spec/models/course/video/session_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe Course::Video::Session do
+  it { is_expected.to belong_to(:submission).inverse_of(:sessions) }
+
+  let!(:instance) { create(:instance, :with_video_component_enabled) }
+  with_tenant(:instance) do
+    describe 'validations' do
+      context 'when session start is after session end' do
+        subject do
+          build(:video_session,
+                session_start: Time.zone.now,
+                session_end: Time.zone.now - 5.minutes)
+        end
+
+        it 'is not valid' do
+          expect(subject).not_to be_valid
+        end
+      end
+
+      context 'when session start is the same as session end' do
+        subject do
+          time = Time.zone.now
+          build(:video_session,
+                session_start: time,
+                session_end: time)
+        end
+
+        it 'is valid' do
+          expect(subject).to be_valid
+        end
+      end
+    end
+
+    describe 'merge_in_events!' do
+      subject do
+        create(:video_session, :with_events)
+      end
+
+      context 'when event already exists' do
+        let!(:time) { Time.zone.now }
+        before do
+          subject.merge_in_events!([{ sequence_num: 1,
+                                      video_time_initial: 2345,
+                                      video_time_final: 10,
+                                      event_type: 'seek',
+                                      event_time: time.midnight }])
+        end
+
+        it 'overwrites the old event ' do
```

Comment:
Random question, when will there be an overwriting of an old event?

Reply:
It supposed to be for a race condition where the same event was sent twice to the server. Technically when this happens the two duplicates contain the same data, but I set overwriting as the default since I expect the frontend to have the most up to date information.